src/spec.c

```diff
@@ -2888,6 +2970,15 @@ int IndexSpec_UpdateDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString
     return REDISMODULE_ERR;
   }
 
+  doc.docExpirationTime.tv_nsec = doc.docExpirationTime.tv_sec = 0;
+  doc.fieldExpirations = NULL;
+  if (spec->monitorDocumentExpiration) {
```

Comment:
Add `__builtin_expect`?

Reply:
I think the indexing process is less in the hot path.
I think it would be better to let the branch prediction do its job.

Reply:
Indexing is in the hot path, I don't think there's a debate on that. But OK let's let the branch prediction work, I don't really know the usage percentages of ttl.